Assets/Scripts/Game/Questing/QuestMachine.cs

```diff
@@ -639,27 +640,33 @@ public Quest ParseQuest(string questName, string[] questSource, int factionId =
         }
 
         /// <summary>
-        /// Parse and instantiate a quest from quest name.
+        /// Parse and start a quest from quest name.
         /// </summary>
         /// <param name="questName">Quest name.</param>
         /// <param name="factionId">Faction id. (optional)</param>
         /// <returns>Quest.</returns>
-        public void InstantiateQuest(string questName, int factionId = 0)
+        public void StartQuest(string questName, int factionId = 0)
         {
             Quest quest = ParseQuest(questName);
             if (quest != null)
             {
                 quest.FactionId = factionId;
-                InstantiateQuest(quest);
+                StartQuest(quest);
             }
         }
 
         /// <summary>
-        /// Instantiate quest from a parsed quest object.
+        /// Start a parsed quest.
         /// </summary>
         /// <param name="quest">Quest.</param>
-        public void InstantiateQuest(Quest quest)
-        {
+        public void StartQuest(Quest quest)
+        {
+            if (!quest.ReadyToStart)
```

Comment:
I'd rather invert the dependency such that the Quest class no longer depends on the TalkManager class, but that we can give the TalkManager a Quest and it can take what it needs from that.
I do agree with not exposing the startTime property like this

Reply:
I see, I'm not very knowledgeable when it comes to DFU quest system, but I know much more on the Talk Manager, so I completely agree with your decoupling proposal. In fact, even in classic sometimes messy code, from what I could decompile of it, talk system isnt't this intermingled with the quest system.